classes/blocks/class-spreadsheet.php

```diff
@@ -51,34 +51,13 @@ public function prepare_data( $fields ): array {
 
 		try {
 			$id = self::extract_sheet_id( $fields['url'] );
-		} catch ( \InvalidArgumentException $exception ) {
-			$fields['error'] = true;
-		}
 
-		if ( isset( $id ) ) {
-
-			$url = "https://docs.google.com/spreadsheets/d/e/${id}/pub?output=csv";
-
-			// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_fopen
-			$handle = fopen( $url, 'rb' );
-
-			if ( false !== $handle ) {
-				$rows = [];
-				while (
-					// phpcs:ignore WordPress.CodeAnalysis.AssignmentInCondition
-					( $data = fgetcsv( $handle, 1000, ',' ) ) !== false
-					// phpcs:ignore Squiz.PHP.DisallowSizeFunctionsInLoops
-					&& count( $rows ) <= self::MAX_ROWS
-				) {
-					$rows[] = $data;
-				}
-				// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_fclose
-				fclose( $handle );
-				$fields['rows'] = $rows;
-			}
+			$sheet = $this->get_sheet( $id, $fields['skip_cache'] ?? true );
```

Comment:
I had added the cache but then realized that if we don't also visualize in the editor that there is a 5 minute delay, or offer a way to invalidate the cache, then this will be confusing to editors setting up the block. So I made it default to true if the block doesn't have the `skip_cache` field so that we can still preserve the logic.